### ISSUE TITLE
Saved excel-file is corrupted by ClosedXML

### ISSUE THREAD

**Issue by v-olin** [ClosedXML/ClosedXML#2121]
## Read and complete the full issue template

Do not randomly delete sections. They are here for a reason.

**Do you want to request a *feature* or report a *bug*?**
- [ ] Bug
- [ ] Feature
- [x] Question

**Did you test against the latest CI build?**
- [ ] Yes
- [x] No

If you answered `No`, please test with [the latest development build](https://ci.appveyor.com/project/ClosedXML/ClosedXML/branch/develop/artifacts) first.

**Version of ClosedXML**

0.102.0

**What is the current behavior?**

My XLWorkbook/excel-file becomes corrupted (and smaller in size?) when saving it.

I would like to save to the same file, but I've also tried to save to a new file but the problem persists.

**What is the expected behavior or new feature?**

That the excel-file is not corrupted.

**Is this a regression from the previous version?**

Same scenario on 0.101.0 and 0.102.0

## Reproducibility
**This is an important section. Read it carefully. Failure to do so will cause a 'RTFM' comment.**

Without a code sample, it is unlikely that your issue will get attention. Don't be lazy. Do the effort and assist the developers to reproduce your problem. Code samples should be [minimal complete and verifiable](https://stackoverflow.com/help/mcve). Sample spreadsheets should be attached whenever applicable. Remove sensitive information.

**Code to reproduce problem:**
```c#
public void Main()
{
    string filepath = @"path of my file";

    var bookStream = new FileStream(filepath, FileMode.Open, FileAccess.ReadWrite, FileShare.ReadWrite);
    IXLWorkbook workbook = new XLWorkbook(bookStream);
    workbook.Save();
}
```
- [x] I attached a sample spreadsheet.  (You can drag files on to this issue)
[EmptyBookPure.xlsx](https://github.com/ClosedXML/ClosedXML/files/11955351/EmptyBookPure.xlsx)

**Comment by jahav** [comment]
File is openable in 0.102 and visuallly looks same. It's not corrupted. 

There is no guarantee nor will ever be that the output file is same as input file. It's a zip full of xml. Output must conform to ISO29500 and can be opened by Excel (Excel has precedence when there is a difference between standard and implementation). That's all.

Whether it is something benign like adding non-implicit namespace to xml or that CLosedXML saves always page setup node (and many other subtle differences).

![image](https://github.com/ClosedXML/ClosedXML/assets/7634052/21789467-21fb-4263-8f3b-6526a326c4e7)

Closing as working as intended.

**Comment by v-olin** [comment]
Maybe the file I gave you was not an appropriate example. Because as you say, the xml-files contain little difference and if I open the saved file in Excel I get a warning that it's corrupted, but Excel is able to repair it. The sheets I'm actually working with are far more complex and cannot be repaired by Excel. But they are confidential so unfortunately I cannot give you an example.

What really annoys me and is the main problem for me is that Excel for any reason at all finds some kind of error with these files I save with ClosedXML. If I take a completely empty excel-file and load it into memory with a `FileStream` and then save it with one of the two way shown below, Excel will **always** tell me that it found a problem with some content. My work is directed to users and I **really** don't want users to have to repair files that's been run through my program.

![Capture](https://github.com/ClosedXML/ClosedXML/assets/55458502/c7aef86a-9bc8-4a03-b11e-1fc3332ba917)

If I click yes on the dialog Excel will open and repair the file without a problem. However, If I try to open one of my actual files Excel will always fail to repair them and they will stay corrupted and unopenable.

FWIW I found a working solution yesterday after work but it's not a very nice solution. I can only avoid corruption if I save it as a different file (but not with a memory stream).

```C#
var fs = new FileStream(@"C:\my\file.xlsx", FileMode.Open, FileAccess.ReadWrite, FileShare.ReadWrite);
var book = new XLWorkbook(fs);

// Corruption
book.Save();

// Corruption
var ts = new FileStream(@"C:\my\output.xlsx", FileMode.CreateNew, FileAccess.ReadWrite, FileShare.ReadWrite);
book.SaveAs(ts);

// Only working way
book.SaveAs(@"C:\my\output.xlsx");
```
